erc1363-function-selector/contracts/PayableTokenReceiver.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.9;
+
+import "erc-payable-token/contracts/token/ERC1363/IERC1363Receiver.sol";
+import "./PayableToken.sol";
+
+contract PayableTokenReceiver is IERC1363Receiver {
+  PayableToken acceptedToken;
+
+  event EventA(address, uint256, uint256);
+  event EventB(address, uint256, bytes3);
+  event EventC(address, uint256, string);
+
+  modifier acceptedTokenOnly() {
+    require(msg.sender == address(acceptedToken), "Accept PayableTokens only");
+    _;
+  }
+  constructor(PayableToken _acceptedToken) {
+    acceptedToken = _acceptedToken;
+  }
+
+  function methodA(address _sender, uint256 _amount, uint256 _value) public acceptedTokenOnly {
```

Comment:
👍 I like that each of the different methods have a different type in the param list!